Stroop task timeline

lakshmanvishnu opened the pull request:
Opened a new branch of my forked repo which only has the stroop task timeline.

lakshmanvishnu committed:
adding stroop task

lakshmanvishnu committed:
adding test file

lakshmanvishnu committed:
updating package-lock.json file

lakshmanvishnu committed:
added additional parameters

lakshmanvishnu committed:
added code for parameters

lakshmanvishnu committed:
debugged parameter

lakshmanvishnu committed:
parameter-related edits

lakshmanvishnu committed:
more parameter-related edits

lakshmanvishnu committed:
more parameter debugging

lakshmanvishnu committed:
adding FNAME and visual reasoning tasks

lakshmanvishnu committed:
remove deleted folder

lakshmanvishnu committed:
adding working version of visual reasoning

lakshmanvishnu committed:
this time for real, visual reasoning working version

lakshmanvishnu committed:
fix: updated instructions and buttons

lakshmanvishnu committed:
fix: standardize the parameter names

lakshmanvishnu committed:
feat: added completed README file

lakshmanvishnu committed:
chore: added docs for stroop task

lakshmanvishnu committed:
Added citations file

lakshmanvishnu committed:
Fix: edited docs

lakshmanvishnu committed:
fix: removing FNAME and visual reasoning test from branch

lakshmanvishnu committed:
Merge remote-tracking branch 'upstream/main'

lakshmanvishnu committed:
Merge branch 'main' into stroop_task

lakshmanvishnu committed:
fix: installing dependancies

lakshmanvishnu committed:
fix: edits to packages and README file

lakshmanvishnu committed:
chore: add changeset

lakshmanvishnu committed:
fix: updating package-lock file

jodeleeuw committed:
remove stroop-specific package-lock.json

lakshmanvishnu committed:
fix: changes to the practice and main trial buttons

jodeleeuw commented on packages/stroop-task/src/style.css:
Does this file get used in your build? I think the `index.html` file is looking for a `style.css` but not finding it because this is in the `/src` directory.

I don't think we should be having separate CSS files that are part of the package, so if we can do without this that's great.

jodeleeuw commented on packages/stroop-task/README.md:
```diff
@@ -0,0 +1,109 @@
+
+## Compatibility
+
+`stroop-timeline` requires jsPsych v8.0.0 or later.
```
this one - previous comment unclear.

jodeleeuw commented on packages/stroop-task/src/index.spec.ts:
there's an array shuffle in `jsPsych.randomization` - we should use that instead so there's not duplicate instances in the codebase

jodeleeuw commented on packages/stroop-task/src/index.spec.ts:
```diff
@@ -0,0 +1,491 @@
+    });
+});
+
+describe("stroop-task timeline components", () => {
```
the tests in here are failing - looks like some of the names may have changed

lakshmanvishnu committed:
fix: edits from code review

lakshmanvishnu committed:
fix: update package-lock.json file

lakshmanvishnu committed:
fix: editing doc files

lakshmanvishnu commented:
add flexible button positioning for main trials

lakshmanvishnu committed:
feat: added text.ts file for instructions

lakshmanvishnu committed:
chore: moving essential styles to index.ts file

jodeleeuw committed:
Merge branch 'stroop_task' of https://github.com/lakshmanvishnu/jspsych-timelines into pr/lakshmanvishnu/69

jodeleeuw committed:
initial cleanup of implementation to remove unnecessary features

jodeleeuw committed:
update text handling in Stroop task

jodeleeuw committed:
some more reconfiguration of the Stroop task, including how text is handled.

jodeleeuw committed:
add results text in text config

jodeleeuw committed:
add docstrings

jodeleeuw committed:
update tests for new implementation

farhat60 commented:
**remaining issue:**
- Buttons go too low on small phones in landscape orientation 
@jadeddelta

jadeddelta committed:
overhaul of css handling, text, and general cleanup

css is now specific to the timeline only, text is
fully customizable, docs are updated and the package
itself is cleaned up.

jadeddelta committed:
refactor config to allow for timeout adjustments, custom colors and fixation

jadeddelta committed:
consolidate docs, simplify configuration options

jodeleeuw commented on packages/stroop-task/src/index.ts:
```diff
@@ -0,0 +1,611 @@
+}
+
+/* Export individual components for custom timeline building */
+export const timelineComponents = {
```
This should be `timelineUnits` for the export.

jodeleeuw commented on packages/stroop-task/src/index.ts:
```diff
@@ -0,0 +1,611 @@
+  // Randomly select the specified number of trials,
+  // selecting each stimulus the same number of times if possible
+  let result: StroopStimulus[] = [];
+  const completeCongruentSets = Math.floor(congruent_trials / congruentStimuli.length);
```
Do we need to guard against divide by 0 if someone puts no congruent or no incongruent trials?

jadeddelta replied:
it is rather the length of congruent/incongruent stimuli rather than no congruent/incongruent trials. it looks like this is tested (handles single color correctly in generalStimuli tests), and is handled as passing 0/0 (NaN) as repeat/sampleWithoutReplacement returns an empty array regardless.

jadeddelta committed:
fix up style export, correct export names, handle 0 stimuli edge case

jadeddelta committed:
remove 0 stimuli edge case, add note

jadeddelta committed:
add parameter for customizing stroop task button html

jadeddelta committed:
standardize text object to use functions instead of placeholders

jadeddelta committed:
fix stroop task button text being highlightable

jadeddelta committed:
fix stroop result type not being exported

jadeddelta committed:
Merge branch 'main' into pr/lakshmanvishnu/69

jadeddelta committed:
re-add yaml by regenerating file under node 18
